app/api/ada/index.js

```diff
@@ -1176,7 +1191,11 @@ export default class AdaApi {
         derivation_scheme: cryptoAccount.derivation_scheme
       };
 
-      const _saveAsAdaAddresses = async _request => {
+      const addresses = [];
+      const indices = [];
+      const _saveAsAdaAddresses = async ({ accountIndex, addrs, offset, addressType}) => {
+        addresses = [...addrs];
+        indices = addrs.map((_, i)=>accountIndex+offset+i);
```

Comment:
I don't understand the logic of adding `accountIndex+offset+i`.
Probably what you need instead is a map from 
```
address => {
    account: number, // accountIndex
    chain: AddressType, // addressType
    index: number, // offset
```
Since this is the information you need to be able to generate the private key for each address given the root private key.